gremlin-client/src/structure/value.rs

```diff
@@ -375,6 +378,36 @@ impl std::convert::TryFrom<GValue> for bool {
     }
 }
 
+impl std::convert::TryFrom<GValue> for f32 {
+    type Error = crate::GremlinError;
+
+    fn try_from(value: GValue) -> GremlinResult<Self> {
+        match value {
+            GValue::Float(x) => Ok(x),
+            GValue::List(s) => from_list(s),
+            _ => Err(GremlinError::Cast(format!(
+                "Cannot cast {:?} to f32",
+                value
+            ))),
+        }
+    }
+}
+
+impl std::convert::TryFrom<GValue> for f64 {
```

Comment:
Just a heads up this is a duplicate implementation of the same code in https://github.com/wolf4ood/gremlin-rs/pull/131/files to keep the PRs independent for review and merging. One of the implementations will need to be removed.